udata/templates/dataset/display.html

```diff
@@ -51,7 +51,7 @@
 {% endblock %}
 
 {% block main_content %}
-<div itemscope="itemscope" itemtype="http://schema.org/Dataset" itemid="{{dataset.id}}">
+<div itemscope="itemscope" itemtype="http://schema.org/Dataset" itemid="{{dataset.id}}" itemslug="{{dataset.slug}}">
```

Comment:
Given the schema: https://schema.org/Dataset I think you should add an extra `<meta>` tag instead:
```html
<meta itemprop="alternateName" content="{{dataset.slug}}" />
```